contracts/RestrictedNORI.sol

```diff
@@ -0,0 +1,1089 @@
+   * proportion of ownership of that schedule's tokens and the total number of tokens being revoked.
+   */
+  function _quantityToRevokePerTokenHolder(
+    uint256 totalQuantityToRevoke,
+    uint256 scheduleId,
+    address account
+  ) private view returns (uint256) {
+    Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];
+    uint256 scheduleTrueTotal = schedule.totalClaimedAmount +
+      totalSupply(scheduleId);
+    // avoid division by 0
+    if (scheduleTrueTotal == 0) {
+      return 0;
+    }
+    uint256 balanceOfAccount = balanceOf(account, scheduleId);
+    uint256 claimedAmountForAccount = schedule.claimedAmountsByAddress[account];
+    uint256 revocableForAccount = ((claimedAmountForAccount +
+      balanceOfAccount) * (totalQuantityToRevoke)) / scheduleTrueTotal;
+
+    return revocableForAccount;
+  }
+
+  /**
+   * @dev The total amount of released tokens available at the current block timestamp for the schedule.
+   * Takes the maximum of either the calculated linearly released amount based on the schedule parameters,
+   * or the released amount floor, which is set at the current released amount whenever the balance of a
+   * schedule is decreased through revocation or withdrawal.
+   */
+  function _releasedBalanceOfSingleSchedule(uint256 scheduleId)
+    internal
+    view
+    returns (uint256)
+  {
+    Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];
+    return
+      MathUpgradeable.max(
+        _linearReleaseAmountAvailable(scheduleId),
+        schedule.releasedAmountFloor
+      );
+  }
+
+  /**
+   * Grants `role` to `account`
+   *
+   * ### Requirements:
+   *    - SCHEDULE_CREATOR_ROLE cannot be granted to an owner of RestrictedNORI tokens
+   *
+   * [See OZ Access Control docs for more] (
+   *  https://docs.openzeppelin.com/contracts/3.x/access-control)
+   */
+  function _grantRole(bytes32 role, address account) internal virtual override {
+    if (role == SCHEDULE_CREATOR_ROLE) {
+      if (_addressToScheduleIdSet[account].length() > 0) {
+        revert RoleUnassignableToScheduleHolder({
+          account: account,
+          role: "SCHEDULE_CREATOR_ROLE"
+        });
+      }
+    }
+    super._grantRole(role, account);
+  }
+
+  // Disabled functions ===========================================
+  /**
+   * Overridden standard ERC1155.setApprovalForAll that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it to revert.
+   */
+  function setApprovalForAll(address, bool) public pure override {
+    revert OperatorActionsNotSupported();
+  }
+
+  /**
+   * Overridden standard ERC1155.isApprovedForAll that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it to revert.
+   */
+  function isApprovedForAll(address, address)
```

Comment:
this doesnt really seem needed since it's pure

Reply:
I think you're probably right... this check will always fail anyway if `setApprovalForAll` is disabled because nobody can ever be approved for anything

Reply:
one potential reason for keeping it is that this way, whenever sometime attempts to transfer someone else's tokens, they get the more informative revert that operator actions are entirely disabled, instead of just hearing that they're not approved... what do you think?